src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1542,6 +1542,12 @@ private BoundExpression InferTypeForDiscardAssignment(BoundDiscardExpression op1
                 {
                     if (isRef)
                     {
+                        // https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#rules-ref-reassignment
+                        // For a ref reassignment in the form ...
+                        // 1. x.e1 = ref e2: where x is safe-to-escape at least return only then e2 must have ref-safe-to-escape at least as large as x
```

Comment:
Not related directly to this PR:
1. Based on this language, I expected the logic below to have some check for `x.e1` situation (member access) in the first branch of the `if` below (line 1553).
2. The spec says "where x is safe-to-escape", but we compute `var leftEscape = GetRefEscape(op1, LocalScopeDepth);`. Are those two equivalent? If so, should the spec say `x.e1 is ref-safe-to-escape ...`, or should the logic below be adjusted to match?
3. The spec says "... at least return-only", but I'm not seeing any such check on LHS in the logic below.

Reply:
From offline discussion, my understanding is that bullet (1) is subsumed by the second half of bullet (2) and therefore the implementation is correct (it matches bullet 2).

Reply:
Filed https://github.com/dotnet/roslyn/issues/64671 for follow-up to simplify the spec. Once done, we can update this comment as well.